### PR TITLE
LPS-140072 Add Done/Cancel buttons to Translation Admin modal

### PR DESCRIPTION
**Pull request by tinycarol**
### References

- [LPS-140072](https://issues.liferay.com/browse/LPS-140072)

### What is the goal?

Add Done/Cancel buttons to Translation Admin modal

### A picture is worth a thousand words

https://user-images.githubusercontent.com/6642927/137287747-30a22e75-32c5-4c59-a525-7e2c18ea68fc.mov

### How to replicate
* Go to Content & Data > Web Content
* Select Add button, choose Basic Web Content
* Click on select translation button (icon with a flag to the left of the title)
* Click on `Manage Translations`
* Click on trash can next to any language
* Click outside the modal/close modal button/cancel button, check the changes have been canceled
* Click on `Manage Translations`
* Add or remove a language
* Click on `Done`, check the changes now show on the select translation dropdown

### Checklist

- ~~[ ] I have made corresponding changes to the documentation~~
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked this works in the corresponding browser

### CODE REVIEW

**Comment by tinycarol** [comment]
ci:test:relevant

File: modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/translation_manager/TranslationAdminModal.js [hunk 1]
```diff
@@ -38,35 +38,30 @@ const TranslationAdminModal = ({
 	const [activeLanguageIds, setActiveLanguageIds] = useState(
 		initialActiveLanguageIds
 	);
-	const [lastDeletedLanguageId, setLastDeletedLanguageId] = useState(null);
 	const [visible, setVisible] = useState(initialVisible);
 
+	const onCancel = () => {
```
Comment (carloslancha) [inline modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/translation_manager/TranslationAdminModal.js:43]: Rename to `handleCancel` to follow same pattern

File: modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/translation_manager/TranslationAdminModal.js [hunk 2]
```diff
@@ -38,35 +38,30 @@ const TranslationAdminModal = ({
...
 		const newActiveLanguageIds = [...activeLanguageIds];
 		newActiveLanguageIds.splice(activeLanguageIds.indexOf(localeId), 1);
 		setActiveLanguageIds(newActiveLanguageIds);
-		setLastDeletedLanguageId(localeId);
 	};
 
-	const handleRestoreLocale = () => {
-		handleAddLocale(lastDeletedLanguageId);
-		setLastDeletedLanguageId(null);
+	const onDoneClicked = () => {
```
Comment (carloslancha) [inline modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/translation_manager/TranslationAdminModal.js:62]: Rename to `handleDone` to follow same pattern

File: modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/translation_manager/TranslationAdminModal.js [hunk 3]
```diff
@@ -38,35 +38,30 @@ const TranslationAdminModal = ({
 	const [activeLanguageIds, setActiveLanguageIds] = useState(
 		initialActiveLanguageIds
 	);
-	const [lastDeletedLanguageId, setLastDeletedLanguageId] = useState(null);
 	const [visible, setVisible] = useState(initialVisible);
 
+	const onCancel = () => {
+		setVisible(false);
+		onClose([...initialActiveLanguageIds]);
```
Comment (carloslancha) [inline modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/translation_manager/TranslationAdminModal.js:45]: Instead calling `onClose` with the `initialActiveLanguageIds` we can first call `setActiveLanguageIds` with the initial values and then call `onClose` with `activeLanguageIds`, in this way we have both values synchronized.

**Review by carloslancha — CHANGES_REQUESTED** [review]
Much better!

Just commented some nits, and remember to sort the methods alphabetically ^^

**Comment by carloslancha** [comment]
@tinycarol  Don't use The Princess Bride against me!

inconceivable-princessbride

**Pull request closed without merging**